bokehjs/src/lib/models/glyphs/segment.ts

```diff
@@ -116,6 +116,18 @@ export class SegmentView extends GlyphView {
     for (const i of candidates) {
       if ((v0[i] <= val && val <= v1[i]) || (v1[i] <= val && val <= v0[i]))
         hits.push(i)
+
+      if (v0[i] == v1[i]) {
+        if (geometry.direction == 'h') {
+          if (Math.abs(this.sx0[i] - sx) <= 2.5) {
+            hits.push(i)
+          }
+        } else {
+          if (Math.abs(this.sy0[i] - sy) <= 2.5) {
```

Comment:
Let's fold this magic constant into `const threshold = 2.5` with a comment explaining the choice of such constant.